geom/type_polygon.go

```diff
@@ -177,7 +189,7 @@ func NewPolygonC(coords [][]Coordinates, opts ...ConstructorOption) (Polygon, er
 			return Polygon{}, err
 		}
 	}
-	return NewPolygon(rings[0], rings[1:], opts...)
+	return NewPolygon(rings, opts...)
```

Comment:
This will result in a double copy of the rings. Could instead separate out the validation into a separate step (which would do no copying internally).